pkg/e2e/e2e.go

```diff
@@ -503,8 +503,16 @@ func cleanupAfterE2E(h *helper.H) (errors []error) {
 	// Do any addon cleanup if configured
 	log.Printf("Addon cleanup: %v", viper.GetBool(config.Addons.RunCleanup))
 	if viper.GetBool(config.Addons.RunCleanup) {
+		// By default, use the existing test harnesses for cleanup
+		harnesses := strings.Split(viper.GetString(config.Addons.TestHarnesses), ",")
+
+		// Check if cleanup harnesses exist and if so, use those instead
+		cleanupHarnesses := viper.GetString(config.Addons.CleanupHarnesses)
+		if len(cleanupHarnesses) > 0 {
+			harnesses = strings.Split(viper.GetString(config.Addons.CleanupHarnesses), ",")
+		}
```

Comment:
Think we need an else here.

Reply:
I don't think we do. I default the cleanup harnesses to be the previously-set test harness, and overwrite `harnesses` only if it's present/valid.

Reply:
Oh, I'm sorry. I overlooked the assign -- yes, please disregard.

Reply:
On the if/else statement, I feel h.RunAddonTests() should possible be move in to so not to have to pass the "cleanup" flag to the clean up harnesses